eks/core.py

```diff
@@ -461,198 +108,233 @@ def compute_initial_guesses(
 
     # Compute temporal differences
     temporal_diffs = ensemble_vars[1:] - ensemble_vars[:-1]
-
     # Compute standard deviation across all temporal differences
     std_dev_guess = round(np.nanstd(temporal_diffs), 5)
     return float(std_dev_guess)
 
 
+def params_nlgssm_for_keypoint(m0, S0, Q, s, R, f_fn, h_fn) -> ParamsNLGSSM:
+    """
+    Construct the ParamsNLGSSM for a single (keypoint) sequence.
+    """
+    return ParamsNLGSSM(
+        initial_mean=jnp.asarray(m0),
+        initial_covariance=jnp.asarray(S0),
+        dynamics_function=f_fn,
+        dynamics_covariance=jnp.asarray(s) * jnp.asarray(Q),
+        emission_function=h_fn,
+        emission_covariance=jnp.asarray(R),
+    )
+
+
 @typechecked
 def optimize_smooth_param(
```

Comment:
let's rename this something else, maybe `run_kalman_smoother`? then we can reserve `optimize_smooth_param` for just that functionality (see below)